src/hdmf/term_set.py

```diff
@@ -12,19 +15,37 @@ class TermSet():
     :ivar view: SchemaView of the term set schema
     """
     def __init__(self,
-                 term_schema_path: str,
+                 term_schema_path: str=None,
+                 schemasheets_folder: str=None,
+                 dynamic: bool=False
                  ):
         """
         :param term_schema_path: The path to LinkML YAML enumeration schema
-
         """
         try:
             from linkml_runtime.utils.schemaview import SchemaView
         except ImportError:
             msg = "Install linkml_runtime"
             raise ValueError(msg)
+
         self.term_schema_path = term_schema_path
-        self.view = SchemaView(self.term_schema_path)
+        self.schemasheets_folder = schemasheets_folder
+
+        if self.schemasheets_folder is not None:
+            if self.term_schema_path is not None:
+                msg = "Cannot have both a path to a Schemasheets folder and a TermSet schema."
+                raise ValueError(msg)
+            else:
+                self.term_schema_path = self.schemasheets_convert()
+                self.view = SchemaView(self.term_schema_path)
+        else:
+            self.view = SchemaView(self.term_schema_path)
+
+        if dynamic:
+            # reset view to now include the dynamically populated term_set
+            self.expanded_term_set_path = self.enum_expander()
+            self.view = SchemaView(self.expanded_term_set_path)
+
```

Comment:
```suggestion
        self.expanded_term_set_path = None
        if dynamic:
            # reset view to now include the dynamically populated term_set
            self.expanded_term_set_path = self.enum_expander()
            self.view = SchemaView(self.expanded_term_set_path)

```

Having the existence of public  object members depend on input parameters is problematic, because it requires users to check if an attribute exists. Instead, we should set `self.expanded_term_set_path` to `None` if it is not being used. If this is strictly and internal variable, then it may also be useful to make the variable private or protected by adding the `_` or `___` prefix.